src/Symfony/Component/Config/Builder/ConfigBuilderGenerator.php

```diff
@@ -0,0 +1,369 @@
+    }
+
+    return $this->PROPERTY;
+}';
+        $class->addMethod($node->getName(), $body, ['PROPERTY' => $property->getName(), 'CLASS' => $childClass->getFqcn()]);
+
+        $this->buildNode($node, $childClass, $this->getSubNamespace($childClass));
+    }
+
+    private function handleVariableNode(VariableNode $node, ClassBuilder $class)
+    {
+        $comment = $this->getComment($node);
+        $property = $class->addProperty($node->getName());
+
+        $body = '
+/**
+COMMENT */
+public function NAME($valueDEFAULT): self
+{
+    $this->PROPERTY = $value;
+
+    return $this;
+}';
+        $class->addMethod($node->getName(), $body, ['PROPERTY' => $property->getName(),  'COMMENT' => $comment, 'DEFAULT' => $node->hasDefaultValue() ? ' = '.var_export($node->getDefaultValue(), true) : '']);
+    }
+
+    private function handlePrototypedArrayNode(PrototypedArrayNode $node, ClassBuilder $class, string $namespace)
+    {
+        $name = $this->getSingularName($node);
+        $prototype = $node->getPrototype();
+        $methodName = $name;
+
+        $parameterType = $this->getParameterType($prototype);
+        if (null !== $parameterType || $prototype instanceof ScalarNode) {
+            $property = $class->addProperty($node->getName());
+            if (null === $key = $node->getKeyAttribute()) {
+                $body = '
+public function NAME(TYPE$value): self
+{
+    $this->PROPERTY = $value;
+
+    return $this;
+}';
+                $class->addMethod($methodName, $body, ['PROPERTY' => $property->getName(), 'TYPE' => '' === $parameterType ? '' : $parameterType.' ']);
+            } else {
+                $body = '
+public function NAME(string $VAR, TYPE$VALUE): self
+{
+    $this->PROPERTY[$VAR] = $VALUE;
+
+    return $this;
+}';
+
+                $class->addMethod($methodName, $body, ['PROPERTY' => $property->getName(), 'TYPE' => '' === $parameterType ? '' : $parameterType.' ', 'VAR' => '' === $key ? 'key' : $key, 'VALUE' => 'value' === $key ? 'data' : 'value']);
+            }
+
+            return;
+        }
+
+        $childClass = new ClassBuilder($namespace, $name);
+        $class->addRequire($childClass);
+        $this->classes[] = $childClass;
+        $property = $class->addProperty($node->getName(), $childClass->getName().'[]');
+
+        if (null === $key = $node->getKeyAttribute()) {
+            $body = '
+public function NAME(array $value = []): CLASS
+{
+    return $this->PROPERTY[] = new CLASS($value);
+}';
+            $class->addMethod($methodName, $body, ['PROPERTY' => $property->getName(), 'CLASS' => $childClass->getFqcn()]);
+        } else {
+            $body = '
+public function NAME(string $VAR, array $VALUE = []): CLASS
+{
+    return $this->PROPERTY[$VAR] ?? $this->PROPERTY[$VAR] = new CLASS($VALUE);
```

Comment:
Maybe I was wrong here, we should fallback to previously registered configuration ONLY when user does not provide a new configuration
As we do in `handleArrayNode`

Reply:
Thank you. I'e added two commits. One with the feature and one with a test.